regnety/dataset/imagenet.py

```diff
@@ -0,0 +1,268 @@
+        Args:
+            example: A dataset example.
+            min_area: Minimum area of image to be used
+            max_iter: Maximum iteration to try before implementing scale and
+                center crop. Randomly genereted width and height must be lower
+                than original width and height before max_iter.
+
+        Returns:
+            Example of same format as TFRECS_FORMAT
+        """
+
+        h, w = tf.cast(example["height"], tf.int64), tf.cast(
+            example["width"], tf.int64
+        )
+        area = h * w
+        return_example = False
+        image = example["image"]
+        num_iter = tf.constant(0, dtype=tf.int32)
+        while num_iter <= max_iter:
+            num_iter = tf.math.add(num_iter, 1)
+            final_area = tf.random.uniform(
+                (), minval=min_area, maxval=1, dtype=tf.float32
+            ) * tf.cast(area, tf.float32)
+            aspect_ratio = tf.random.uniform(
+                (), minval=3.0 / 4.0, maxval=4.0 / 3.0, dtype=tf.float32
+            )
+            w_cropped = tf.cast(
+                tf.math.round(tf.math.sqrt(final_area * aspect_ratio)), tf.int64
+            )
+            h_cropped = tf.cast(
+                tf.math.round(tf.math.sqrt(final_area / aspect_ratio)), tf.int64
+            )
+
+            if tf.random.uniform(()) < 0.5:
+                w_cropped, h_cropped = h_cropped, w_cropped
+
+            if h_cropped <= h and w_cropped <= w:
+                return_example = True
+                if h_cropped == h:
+                    y = tf.constant(0, dtype=tf.int64)
+                else:
+                    y = tf.random.uniform(
+                        (), minval=0, maxval=h - h_cropped, dtype=tf.int64
+                    )
+                if w_cropped == w:
+                    x = tf.constant(0, dtype=tf.int64)
+                else:
+                    x = tf.random.uniform(
+                        (), minval=0, maxval=w - w_cropped, dtype=tf.int64
+                    )
+
+                image = image[y : (y + h_cropped), x : x + w_cropped, :]
+                image = tf.image.resize(
+                    image, (self.image_size, self.image_size)
+                )
+                break
+        if return_example:
+            return {
+                "image": image,
+                "height": self.image_size,
+                "width": self.image_size,
+                "filename": example["filename"],
+                "label": example["label"],
+                "synset": example["synset"],
+            }
+
+        image = self._scale_and_center_crop(
+            image, h, w, self.image_size, self.image_size
+        )
+        return {
+            "image": image,
+            "height": self.image_size,
+            "width": self.image_size,
+            "filename": example["filename"],
+            "label": example["label"],
+            "synset": example["synset"],
+        }
+
+    def clip_example(self, example):
```

Comment:
Why is this called "clip_example"? The name doesn't make sense to me. Maybe "one_hot_encode_example"?

Reply:
I'll change it